yrs/src/store.rs

```diff
@@ -152,7 +152,8 @@ impl Store {
             encoder.write_uvar(clock);
             let first_block = blocks.get(start);
             // write first struct with an offset
-            first_block.encode(Some(self), encoder);
+            let offset = clock - first_block.id().clock;
+            first_block.encode_with_offset(Some(self), encoder, offset);
```

Comment:
Fix 3: when encoding given state vector, first block must be encoded with offset, because incoming state vector clock may point inside of it.